front/pages/api/w/[wId]/data_sources/[dsId]/files.ts

```diff
@@ -0,0 +1,126 @@
+import type { FileType, WithAPIErrorResponse } from "@dust-tt/types";
+import type { NextApiRequest } from "next";
+import type { NextApiResponse } from "next";
+
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/auth_wrappers";
+import type {
+  UpsertDocumentArgs,
+  UpsertTableArgs,
+} from "@app/lib/api/data_sources";
+import { getOrCreateConversationDataSourceFromFile } from "@app/lib/api/data_sources";
+import { processAndUpsertToDataSource } from "@app/lib/api/files/upsert";
+import type { Authenticator } from "@app/lib/auth";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { FileResource } from "@app/lib/resources/file_resource";
+import { apiError } from "@app/logger/withlogging";
+
+export interface UpsertFileToDataSourceRequestBody {
+  fileId: string;
+  upsertArgs:
+    | Pick<UpsertDocumentArgs, "name" | "title" | "tags">
+    | Pick<
+        UpsertTableArgs,
+        | "name"
+        | "title"
+        | "description"
+        | "tableId"
+        | "tags"
+        | "useAppForHeaderDetection"
+      >;
+}
+
+export interface UpsertFileToDataSourceResponseBody {
+  file: FileType;
+}
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<{ file: FileType }>>,
+  auth: Authenticator
+): Promise<void> {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
+      },
+    });
+  }
+
+  const { fileId, upsertArgs } = req.body;
+
+  // Get file and make sure that it is within the same workspace.
+  const file = await FileResource.fetchById(auth, fileId);
+  if (!file) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "file_not_found",
+        message: "File not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST": {
+      let dataSourceToUse: DataSourceResource | null = null;
+      // JIT data source.
+      if (file.useCase === "conversation") {
```

Comment:
@Fraggle as reviewer. Is this used? That does not make a ton of sense given the dataSource is specified in the path??